salt/matchers/pillar_exact_match.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+'''
+This is the default pillar exact matcher.
+
+NOTE: These functions are converted to methods on the Matcher class during master and minion startup.
+This is why they all take `self` but are not defined inside a `class:` declaration.
+'''
+from __future__ import absolute_import, print_function, unicode_literals
+
+import logging
+from salt.ext import six  # pylint: disable=3rd-party-module-not-gated
+from salt.defaults import DEFAULT_TARGET_DELIM  # pylint: disable=3rd-party-module-not-gated
+import salt.utils.data  # pylint: disable=3rd-party-module-not-gated
+
+log = logging.getLogger(__name__)
+
+
+def match(self, tgt, delimiter=':'):
+    '''
+    Reads in the pillar match, no globbing, no PCRE
+    '''
+    log.debug('pillar target: %s', tgt)
+    if delimiter not in tgt:
+        log.error('Got insufficient arguments for pillar match '
+                  'statement from master')
```

Comment:
This is an insufficient message either. :wink: Because it doesn't give any hints what to do and even where it comes from, unless you know where it is and know how to read this code. Can we rephrase this and give at least hint how to fix it?

Reply:
I went through all these and improved the messages.